### PR TITLE
coloring: Only generate strictly positive IDs

### PR DESCRIPTION
**Pull request by jcbrinfo**
This will allow to reserve negative IDs (including 0) for special purposes.

Signed-off-by: Jean-Christophe Beaupré <jcbrinfo@users.noreply.github.com>

### CODE REVIEW

File: src/compiler/coloring.nit [hunk 1]
```diff
@@ -160,9 +160,16 @@ end
 #
 # Possible colors:
 #
-# * A:0, B:1, C: 2, D: 1, E: 3, F:3, G:2, H:4
+# * A: 0
```
Comment (privat) [inline src/compiler/coloring.nit]: If you start at 0, you fail your own PR

Comment (privat) [inline src/compiler/coloring.nit]: If you start at 0, you fail your own PR

Comment (jcbrinfo) [inline src/compiler/coloring.nit]: These are *colors* (see above). My PR is about IDs.

**Comment by jcbrinfo** [comment]
IDs are never used as indices (only colors are). So, there is nothing that avoid us to start IDs at 1. Also, the only client of that interface is the compiler (in particular, `separate_compiler` and `separate_erasure_compiler`). Then, changing the service instead of the clients simply avoid repeating the same code twice.

**Comment by privat** [comment]
OK. I had mistaken colors with id.

**Comment by jcbrinfo** [comment]
I knew that. 😉

**Pull request merged**